### PR TITLE
[SYCL][ESIMD] Disable inline for dgetrf.cpp for esimd_emulator otherwise host compile will hang

### PR DESCRIPTION
**Pull request by yubingex007-a11y**

### CODE REVIEW

File: sycl/test-e2e/ESIMD/regression/dgetrf.cpp [hunk 1]
```diff
@@ -9,7 +9,7 @@
 // REQUIRES: gpu
 // UNSUPPORTED: gpu-intel-gen9 && windows
 // UNSUPPORTED: cuda || hip
-// RUN: %{build} -I%S/.. -o %t.out
+// RUN: %{build} -I%S/.. -O1 -fno-inline-functions -o %t.out
```
Comment (v-klochkov) [inline sycl/test-e2e/ESIMD/regression/dgetrf.cpp:12]: no-inline seems Ok for this test that gives timeout because of that.
But why -O1 is needed here?

**Comment by yubingex007-a11y** [comment]
ping?

**Review by sarnex — APPROVED** [review]
manually confirmed we don't need `-O1` and just `-fno-inline-functions` fixes it

**Comment by dm-vodopyanov** [comment]
```
Failed Tests (1):
  SYCL :: ESIMD/regression/dgetrf.cpp
```

PR can't be merged. @yubingex007-a11y, please analyze and fix the failure.

**Comment by sarnex** [comment]
@yubingex007-a11y Looks like CI fails on this test with this change:

```
$ "env" "ONEAPI_DEVICE_SELECTOR=opencl:gpu" "/__w/llvm/llvm/build-e2e/ESIMD/regression/Output/dgetrf.cpp.tmp.out" "3" "2" "1"
# command stderr:
warning: TransformPrivMem: _ZN4sycl3_V13ext5intel5esimd6detail13simd_obj_implIdLi384ENS3_4simdIdLi384EEEvE20cast_this_to_derivedEv_direct allocation size is too big: using TPM
warning: TransformPrivMem: _ZN4sycl3_V13ext5intel5esimd6detail14simd_view_implINS3_4simdIdLi384EEENS3_11region_baseILb0EdLi1ELi1ELi64ELi1EEEEC2ERS7_S9__direct allocation size is too big: using TPM

...

error: command failed with exit status: -11
```

So it seems the GPU requires the inlining.

We can split the test into a gpu version and an emu version, see something I did recently here: https://github.com/intel/llvm/commit/6a13e45c5abd3f3d74d81efba23e1578095afd67

**Comment by yubingex007-a11y** [comment]
i have little knowledge of gpu, @sarnex  can you take a look at the fail?
i might have a guess: it is because we change the device code by disable inlining.
i think we should pass fno-inline to host code only. i will modify the patch.

**Comment by sarnex** [comment]
@yubingex007-a11y i think the gpu fail is understood, the warning tells us the problem. 

yeah, change host only.

**Review by v-klochkov — APPROVED** [review]
This test case looks like a good example for creation of a new issue/tracker - inliner must not go super crazy and lead to hanging, it should have some heuristics to limit itself.

**Comment by v-klochkov** [comment]
The test fails on Linux/Gen9.

**Comment by sarnex** [comment]
@yubingex007-a11y btw i just tried testing this and found something important. `-fsycl-host-compiler-options` has no effect unless `-fsycl-host-compiler` is set. so you will need to set `-fsycl-host-compiler` too.

**Comment by yubingex007-a11y** [comment]
> @yubingex007-a11y btw i just tried testing this and found something important. `-fsycl-host-compiler-options` has no effect unless `-fsycl-host-compiler` is set. so you will need to set `-fsycl-host-compiler` too.

i have no issue when -fsycl-host-compiler=**icpx** -fsycl-host-compiler-options="-O2 -fno-inline-functions -Wno-unusable-partial-specialization"
but i have ld linker issue when when -fsycl-host-compiler=**clang** -fsycl-host-compiler-options="-O2 -fno-inline-functions -Wno-unusable-partial-specialization"

**Comment by yubingex007-a11y** [comment]
besides, i saw we've disabled the testcase for intel-gen9
// UNSUPPORTED: gpu-intel-gen9
it should be filtered out in gen9's testing, right? @dm-vodopyanov

**Comment by sarnex** [comment]
@yubingex007-a11y no we can't use icpx here. 

this is getting too complicated, can we just split the test like i did in https://github.com/intel/llvm/commit/6a13e45c5abd3f3d74d81efba23e1578095afd67?

for emu case, we just pass ` -fno-inline-functions` to both host and device compiler

for gpu case, no changes from today

**Comment by yubingex007-a11y** [comment]
@steffenlarsen can you help us merge it? the pre-ci fail is not related

**Pull request merged**